dev-packages/cli/src/commands/generate-index.ts

```diff
@@ -0,0 +1,183 @@
+            writeIndex(directory, children, options);
+        }
+    }
+}
+
+export function isDirectChild(parent: string, child: string, childHasChildren: () => boolean): boolean {
+    return isChildFile(parent, child) || (isChildDirectory(parent, child) && childHasChildren());
+}
+
+export function isDirectory(file: string): boolean {
+    return file.endsWith('/');
+}
+
+export function isFile(file: string): boolean {
+    return !isDirectory(file);
+}
+
+export function getLevel(file: string): number {
+    return file.split('/').length;
+}
+
+export function isChild(parent: string, child: string): boolean {
+    return child.startsWith(parent);
+}
+
+export function isChildDirectory(parent: string, child: string): boolean {
+    return isDirectory(child) && isChild(parent, child) && getLevel(child) === getLevel(parent) + 1;
+}
+
+export function isChildFile(parent: string, child: string): boolean {
+    return isFile(child) && isChild(parent, child) && getLevel(child) === getLevel(parent);
+}
+
+export function writeIndex(directory: string, exports: string[], options: GenerateIndexCmdOptions): void {
+    const indexFile = path.join(process.cwd(), directory, 'index.ts');
+    if (exports.length === 0) {
+        if (options.forceOverwrite && fs.existsSync(indexFile)) {
+            console.log('Remove index file', indexFile);
+            fs.rmSync(indexFile);
+        }
+        return;
+    }
+    const exists = fs.existsSync(indexFile);
+    if (exists && !options.forceOverwrite) {
+        console.log("Do not overwrite existing index file. Use '-f' to force an overwrite.", indexFile);
+        return;
+    }
+
+    const headerContent = exists ? extractHeader(fs.readFileSync(indexFile, { encoding: 'utf-8' })) : defaultHeader();
+    const exportContent = exports.map(exported => createExport(directory, exported, options)).sort();
+    const content = [...headerContent, ...exportContent].join(os.EOL) + os.EOL; // end with an empty line
+    console.log((exists ? 'Overwrite' : 'Write') + ' index file', indexFile);
+    verbose(options, () => '  ' + content.split(os.EOL).join(os.EOL + '  '));
+    fs.writeFileSync(indexFile, content, { flag: 'w' });
+}
+
+export function createExport(directory: string, relativePath: string, options: GenerateIndexCmdOptions): string {
+    // remove directory prefix, file extension and directory ending '/'
+    const parentPrefix = directory.length;
+    const suffix = isFile(relativePath) ? path.extname(relativePath).length : 1;
+    const relativeName = relativePath.substring(parentPrefix, relativePath.length - suffix);
+    const exportName = options.style === 'esm' && isFile(relativePath) ? relativeName + '.js' : relativeName;
+    const exportLine = `export * from './${exportName}';`;
+    return exportLine;
+}
+
+export function verbose(options: GenerateIndexCmdOptions, message?: any, ...optionalParams: any[]): void {
+    if (options.verbose) {
```

Comment:
Any particular reason why you don`t use the `LOGGER` utility we already have in place for this?

Just configure the logger in your entry function with
```ts
    configureLogger(options.verbose);
  //    and use it with 
   LOGGER.info() // for non-verbose logging
   LOGGER.debug() // for verbose logging
```

see also (https://github.com/eclipse-glsp/glsp/blob/b5aff4fd6b78bf4c61ce7190cc4a84a0d06626aa/dev-packages/cli/src/commands/update-next.ts#L35)

Reply:
The reason is that I just wasn't properly aware of it, I'll fix it.